enhancements/network/user-defined-network-segmentation.md

```diff
@@ -0,0 +1,956 @@
+endpoints are those IP addresses which reside on the user-defined primary network. In order to solve this problem,
+OVN-Kubernetes may create its own endpoint slices or may choose to do dynamic lookups at runtime to map endpoints to
+their primary IP address. Leveraging a second set of endpoint slices will be the preferred method, as it creates less
+indirection and gives explicit Kube API access to what IP addresses are being used by OVN-Kubernetes.
+
+Kubelet health checks to pods are queried via the cluster default network. When endpoints are considered unhealthy they
+will be removed from the endpoint slice, and thus their primary IP will be removed from the OVN load balancer. However,
+it is important to note that the healthcheck is being performed via the cluster default network interface on the pod, 
+which ensures the application is alive, but does not confirm network connectivity of the primary interface. Therefore,
+there could be a situation where OVN networking on the primary interface is broken, but the default interface continues
+to work and reports 200 OK to Kubelet, thus rendering the pod serving in the endpoint slice, but unable to function.
+Although this is an unlikely scenario, it is good to document.
+
+### Network Policy
+
+Configuring network policies that allow traffic between user-defined network namespaces will not be invalidated or
+ignored by OVN-Kubernetes. If a user configures a network policy that would have no effect, (like stopping namespace A
+from talking to B on different networks) then OVN-Kubernetes will still configure this policy in OVN. These policies
+will also have no effect between namespaces that are assigned to different networks as there will be no network
+connectivity between the namespaces. However, there may still be a desire to configure network policy between namespaces
+that share the same network, as well as for other reasons like ingress/egress control. Additionally, policies that
+block/allow cluster egress or ingress traffic will still be enforced.
+
+### API Extensions
+
+The main API extension here will be a namespace scoped network CRD as well a cluster scoped network CRD. These CRDs
+will be registered by Cluster Network Operator (CNO). See the [CRDs for Managing Networks](#crds-for-managing-networks)
+section for more information on how the CRD will work. There will be a finalizer on the CRDs, so that upon deletion
+OVN-Kubernetes can validate that there are no pods still using this network. If there are pods still attached to this
+network, the network will not be removed.
+
+### Workflow Description
+
+#### Tenant Use Case
+
+As a tenant I want to ensure when I create pods in my namespace their network traffic is isolated from other tenants on
+the cluster. In order to ensure this, I first create a network CRD that is namespace scoped and indicate:
+
+ - Type of network (Layer 3 or Layer 2)
+ - IP addressing scheme I wish to use (optional)
```

Comment:
not clear what this means; I don't think you've mentioned anything about that previously

Reply:
In the user stories I mentioned it:
```
* As a user, I want to be able to use my own consistent IP addressing scheme in my network. I want to be able to specify
  and re-use the same IP subnet for my pods across different namespaces and clusters. This provides a consistent
  and repeatable network environment for administrators and users.
```

Reply:
ah, I thought "scheme" implied something like "I want to use DHCP on this network", not just specifying a CIDR range for ovn-k to use on this network.